File: app/src/main/java/ch/threema/app/activities/LicenseActivity.java
/*  _____ _
 * |_   _| |_  _ _ ___ ___ _ __  __ _
 *   | | | ' \| '_/ -_) -_) '  \/ _` |_
 *   |_| |_||_|_| \___\___|_|_|_\__,_(_)
 *
 * Threema for Android
 * Copyright (c) 2013-2025 Threema GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package ch.threema.app.activities;

import android.os.Bundle;

import org.slf4j.Logger;

import ch.threema.app.R;
import ch.threema.base.utils.LoggingUtil;

import static ch.threema.app.utils.ActiveScreenLoggerKt.logScreenVisibility;

public class LicenseActivity extends SimpleWebViewActivity {
    private static final Logger logger = LoggingUtil.getThreemaLogger("LicenseActivity");


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        logScreenVisibility(this, logger);
    }

    @Override
    protected int getWebViewTitle() {
        return R.string.os_licenses;
    }

    @Override
    protected String getWebViewUrl() {
        return "file:///android_asset/license.html";
    }

    @Override
    protected boolean requiresConnection() {
        return false;
    }
}

File: app/src/main/java/ch/threema/app/webviews/LicenseActivity.kt
package ch.threema.app.webviews

import android.content.Context
import ch.threema.android.buildActivityIntent
import ch.threema.app.R
import ch.threema.app.utils.logScreenVisibility
import ch.threema.base.utils.getThreemaLogger

private val logger = getThreemaLogger("LicenseActivity")

class LicenseActivity : SimpleWebViewActivity() {
    init {
        logScreenVisibility(logger)
    }

    override fun getWebViewTitle() = R.string.os_licenses

    override fun getWebViewUrl(isDarkTheme: Boolean) = "file:///android_asset/license.html"

    override fun requiresConnection() = false

    companion object {
        fun createIntent(context: Context) = buildActivityIntent<LicenseActivity>(context)
    }
}
